nilearn/glm/first_level/first_level.py

```diff
@@ -1155,10 +1164,10 @@ def _check_events_file_uses_tab_separators(events_files):
 
 def _check_run_tables(run_imgs, tables_, tables_name):
     """Check fMRI runs and corresponding tables to raise error if necessary."""
-    if isinstance(tables_, (str, pd.DataFrame, np.ndarray)):
+    if isinstance(tables_, (str, Path, pd.DataFrame, np.ndarray)):
```

Comment:
Should np.ndarray really be here?

Reply:
well check_and_load_tables can in theory accept arrays from my reading of the code so I think the original intent was trying to stick to that

but you are correct that we don't allow it explicitly as an input so we probably should not check for it

Reply:
OK that's because confounds in FirstLevelModel.fit can be a numpy array and we also use this function to check it

Reply:
ok something does not make sense about allowing numpy array: they are accepted as inputs but never appended to the output list... so they will never be passed on to whatever needs them later...

I think that in a follow up PR I will check if we actually support array as confounds and it is properly implemented or if this is a doc string error.

@bthirion @emdupre 
I may sound like a broken record but with type annotations this kind of headache would happen a lot less often.